### PR TITLE
1398 term definition annotator select

### PR DESCRIPTION
**Pull request by ledsoft**
https://kbss.felk.cvut.cz/redmine/issues/1398

Note that to work correctly, this PR requires the backend changes (https://github.com/kbss-cvut/termit/pull/34) to be merged and deployed.

### CODE REVIEW

File: src/component/annotator/Annotator.tsx [hunk 1]
```diff
@@ -148,38 +154,62 @@ export class Annotator extends React.Component<AnnotatorProps, AnnotatorState> {
...
-        } else {
-            // TODO Creating occurrences is not implemented, yet
-            return Promise.resolve();
         }
+        // TODO Creating occurrences is not implemented, yet
+    }
+
+    public onSaveTermDefinition = (term: Term) => {
+        this.setTermDefinitionSource(term, this.state.existingTermDefinitionAnnotationElement!)
+            .then(() => {
```
Comment (psiotwo) [inline src/component/annotator/Annotator.tsx]: { return ... can be omitted

File: src/component/term/TermMetadataEdit.tsx [hunk 2]
```diff
@@ -85,18 +85,11 @@ export class TermMetadataEdit extends React.Component<TermMetadataEditProps, Ter
         this.setState({scopeNote: Object.assign({}, this.state.scopeNote, change)});
     };
 
-    public onDefinitionChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-        const value = e.currentTarget.value;
-        const change = {};
-        change[this.props.language] = value;
-        this.setState({definition: Object.assign({}, this.state.definition, change)});
+    public onChange = (change: Partial<TermData>) => {
+        // @ts-ignore
```
Comment (psiotwo) [inline src/component/term/TermMetadataEdit.tsx:89]: Why this?

**Review by psiotwo — APPROVED** [review]
lgtm

File: src/component/annotator/Annotator.tsx (hunk 1 above)
Comment (ledsoft) [reply src/component/annotator/Annotator.tsx]: Good point.

File: src/component/term/TermMetadataEdit.tsx (hunk 2 above)
Comment (ledsoft) [reply src/component/term/TermMetadataEdit.tsx:89]: Won't compile without the ignore. Partial marks all properties optional, but label is not in the state.

**Pull request merged**